File: java/src/main/java/com/alxarafe/app/application/catalogue/ItemFamilyRepository.java
package com.alxarafe.app.application.catalogue;

import com.alxarafe.app.domain.catalogue.entity.ItemFamily;
import com.alxarafe.app.domain.catalogue.valueobject.ItemFamilyCode;
import com.alxarafe.app.domain.rules.valueobject.AttributeCode;

import java.util.List;
import java.util.Optional;

public interface ItemFamilyRepository {
    Optional<ItemFamily> findByCode(ItemFamilyCode code);

    List<ItemFamily> findAll();

    List<String> availableFamilyAttributes(List<AttributeCode> codes);

    void save(ItemFamily family);
}

File: php/src/Application/Catalogue/ItemFamilyRepository.php
<?php

declare(strict_types=1);

namespace Alxarafe\App\Application\Catalogue;

use Alxarafe\App\Domain\Catalogue\Entity\ItemFamily;
use Alxarafe\App\Domain\Catalogue\ValueObject\ItemFamilyCode;
use Alxarafe\App\Domain\Catalogue\ValueObject\StorageAttributeId;

interface ItemFamilyRepository
{
    public function findByCode(ItemFamilyCode $code): ?ItemFamily;

    /** @return list<ItemFamily> */
    public function findAll(): array;

    /** @param list<StorageAttributeId> $ids
     *  @return list<string>
     */
    public function existingAttributeIds(array $ids): array;

    /** Guarda familia y vínculos atómicamente; revalida y protege las referencias durante el guardado. */
    public function save(ItemFamily $family): void;
}
